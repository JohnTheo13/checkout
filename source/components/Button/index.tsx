import React, { ButtonHTMLAttributes, ReactElement } from 'react'

import ButtonWrapper from './styles'

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  title: string
}

const Button = ({ title, ...rest }: ButtonProps): ReactElement => (
  <ButtonWrapper {...rest}>
    {title}
  </ButtonWrapper>
)

export default Button
