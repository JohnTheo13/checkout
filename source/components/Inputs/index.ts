export { default as Input } from './Base'
export { default as InputError } from './InputError'
export { default as CardInput } from './CardInput'
