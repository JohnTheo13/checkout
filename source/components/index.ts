export { default as Svg, IconName } from './Svg'
export { default as PaymentOption, Option } from './PaymentOption'
export { default as Mollie } from './Mollie'
export { default as Button } from './Button'
