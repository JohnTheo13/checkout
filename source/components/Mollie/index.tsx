import React, { ReactElement } from 'react'

import Container from './style'
import Svg, { IconName } from '../Svg'

const Mollie = (): ReactElement => (
  <Container>
    <Svg name={IconName.lock} />
    <div>Payment secured and powered by <span>mollie</span></div>
  </Container>
)

export default Mollie
