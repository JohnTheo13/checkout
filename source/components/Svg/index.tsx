import React, { ReactElement } from 'react'
import InputError from './icons/input-error.svg'
import Lock from './icons/lock.svg'
import Amex from './logos/amex.svg'
import BankTransfer from './logos/banktransfer.svg'
import CreditCard from './logos/creditcard.svg'
import Ideal from './logos/ideal.svg'
import MasterCard from './logos/mastercard.svg'
import Mollie from './logos/mollie-logo.svg'
import Merchant from './logos/merchant-logo.svg'
import Visa from './logos/visa.svg'

export enum IconName {
  lock = 'lock',
  inputError = 'inputError',
  amex = 'amex',
  bankTransfer = 'bankTransfer',
  creditCard = 'creditCard',
  ideal = 'ideal',
  masterCard = 'masterCard',
  mollie = 'mollie',
  merchant = 'merchant',
  visa = 'visa'
}

const Svg = ({ name }: { name: IconName }): ReactElement => {
  switch (name) {
    case IconName.lock:
      return <Lock />
    case IconName.inputError:
      return <InputError />
    case IconName.amex:
      return <Amex />
    case IconName.bankTransfer:
      return <BankTransfer />
    case IconName.creditCard:
      return <CreditCard />
    case IconName.ideal:
      return <Ideal />
    case IconName.visa:
      return <Visa />
    case IconName.mollie:
      return <Mollie />
    case IconName.merchant:
      return <Merchant />
    case IconName.masterCard:
      return <MasterCard />
  }
}

export default Svg
