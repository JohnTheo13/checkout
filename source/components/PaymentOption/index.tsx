import React, { ReactElement } from 'react'

import { ButtonWrapper } from './styles'
import Svg, { IconName } from '../Svg'

const { creditCard, ideal, bankTransfer, ...RestNames } = IconName
type excludedOptions = keyof typeof RestNames
export type Option = Exclude<IconName, excludedOptions>

const title = {
  [creditCard]: 'Credit card',
  [ideal]: 'iDEAL',
  [bankTransfer]: 'Bank Transfer'
}
const ButtonLink = ({ name, ...rest }: { name: Option, onClick: () => void }): ReactElement => (
  <ButtonWrapper {...rest} >
    <div>
      <Svg name={name} />
      <span>{title[name]}</span>
    </div>
    <span>{'>'}</span>
  </ButtonWrapper>
)

export default ButtonLink
