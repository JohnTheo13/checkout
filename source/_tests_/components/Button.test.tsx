import React from 'react'
import renderer from 'react-test-renderer'

import { Button } from '../../components'

test('Buttton snapshot', () => {
  const click = jest.fn()
  const title = 'title'
  const button = renderer.create(<Button title={title} onClick={click} />)
  expect(button.root.findByType('button').props.children).toBe(title)
  const inst = button.root.findByType('button')
  inst.props.onClick()
  expect(click).toBeCalled()
  expect(button.toJSON()).toMatchSnapshot()
})
