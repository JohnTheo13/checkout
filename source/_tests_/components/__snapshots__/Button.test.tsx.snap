// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttton snapshot 1`] = `
<button
  className="sc-dlfnbm hedudr"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  title
</button>
`;
